odk/src/test/java/org/motechproject/odk/event/builder/impl/EventBuilderOdkTest.java

```diff
@@ -0,0 +1,139 @@
+
+        event = itr.next();
+        assertEquals(event.getSubject(),"org.motechproject.odk.repeat_group.configName.ona_nested_repeats.outer_group/inner_group");
+        assertEquals(event.getParameters().size(),5);
+        assertEquals(event.getParameters().get("root_scope_field"),"root scope");
+        assertEquals(event.getParameters().get("outer_group/outer_group_field"),"outer group 2 field");
+        assertEquals(event.getParameters().get("formhub/uuid"),"517b7383ae62455482b6f8505e5646e2");
+        assertEquals(event.getParameters().get("meta/instanceID"),"uuid:d9e30fd7-5f0c-4185-a331-db0f740b3547");
+        assertEquals(event.getParameters().get("outer_group/inner_group/inner_group_field"),"outer group 2 inner group field 1");
+
+        event = itr.next();
+        assertEquals(event.getSubject(),"org.motechproject.odk.repeat_group.configName.ona_nested_repeats.outer_group/inner_group");
+        assertEquals(event.getParameters().size(),5);
+        assertEquals(event.getParameters().get("root_scope_field"),"root scope");
+        assertEquals(event.getParameters().get("outer_group/outer_group_field"),"outer group 2 field");
+        assertEquals(event.getParameters().get("formhub/uuid"),"517b7383ae62455482b6f8505e5646e2");
+        assertEquals(event.getParameters().get("meta/instanceID"),"uuid:d9e30fd7-5f0c-4185-a331-db0f740b3547");
+        assertEquals(event.getParameters().get("outer_group/inner_group/inner_group_field"),"outer group 2 inner group field 2");
+
+        event = itr.next();
+        assertEquals(event.getSubject(),"org.motechproject.odk.recieved_form.configName.ona_nested_repeats");
+        assertEquals(event.getParameters().size(),6);
+        assertEquals(event.getParameters().get("root_scope_field"),"root scope");
+        assertEquals(event.getParameters().get("meta/instanceID"),"uuid:d9e30fd7-5f0c-4185-a331-db0f740b3547");
+        assertEquals(event.getParameters().get("formhub/uuid"),"517b7383ae62455482b6f8505e5646e2");
+        assertEquals(event.getParameters().get("root_scope_field"),"root scope");
+        assertEquals(event.getParameters().get(EventParameters.CONFIGURATION_NAME),configuration.getName());
+        assertEquals(event.getParameters().get(EventParameters.FORM_TITLE),formDefinition.getTitle());
+        assertNotNull(event.getParameters().get("outer_group"));
+
+    }
+
+
+    private void alterFormDef(FormDefinition formDefinition) {
+        List<FormElement> formElements = formDefinition.getFormElements();
+
+        for (FormElement formElement : formElements) {
+            if (!formElement.isPartOfRepeatGroup()) {
+                alterFormElementName(formElement);
+            }
+        }
+    }
+
+    private void alterFormElementName(FormElement formElement) {
+        String formFieldName = formElement.getName();
+        String name = formFieldName.substring(formFieldName.indexOf("/", 1) + 1, formFieldName.length()); // removes form title from URI
```

Comment:
You know what - I have seen this before, multiple times. This index manipulating belongs in some kind of utility class (example String FormUtils.removeTitleFromURI(String uri)).